modules/pricer/src/main/java/com/opengamma/strata/pricer/sensitivity/CurveSensitivityUtils.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.sensitivity;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.common.collect.ImmutableList;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.market.curve.CurveCurrencyParameterSensitivities;
+import com.opengamma.strata.market.curve.CurveCurrencyParameterSensitivity;
+import com.opengamma.strata.market.curve.CurveMetadata;
+import com.opengamma.strata.market.curve.CurveParameterMetadata;
+import com.opengamma.strata.market.curve.DatedCurveParameterMetadata;
+import com.opengamma.strata.market.curve.DefaultCurveMetadata;
+import com.opengamma.strata.market.curve.meta.SimpleCurveNodeMetadata;
+import com.opengamma.strata.market.curve.meta.TenorCurveNodeMetadata;
+
+/**
+ * Utilities to transform sensitivities.
+ */
+public class CurveSensitivityUtils {
+
+  /**
+   * Re-buckets a {@link CurveCurrencyParameterSensitivities} to a given set of dates. 
+   * <p>
+   * The list of dates must be sorted in chronological order. All sensitivities are re-bucketed to the same date list.
+   * The re-bucketing is done by linear weighting on the number of days, i.e. the sensitivities for dates outside the 
+   * extremes are fully bucketed to the extremes and for date between two re-bucketing dates, the weight on the start 
+   * date is the number days between end date and the date re-bucketed divided by the number of days between the 
+   * start and the end.
+   * The input sensitivity should have a {@link DatedCurveParameterMetadata} for each sensitivity. 
+   * 
+   * @param sensitivities  the input sensitivities
+   * @param targetDates  the list of dates for the re-bucketing
+   * @return the sensitivity after the re-bucketing
+   */
+  public static CurveCurrencyParameterSensitivities linearRebucketing(
+      CurveCurrencyParameterSensitivities sensitivities,
+      List<LocalDate> targetDates) {
```

Comment:
The dates should probably be checked to ensure they are sorted. Or maybe just copy and sort them. Will an unsorted list cause the return value to be invalid? We shouldn't allow that to happen silently.